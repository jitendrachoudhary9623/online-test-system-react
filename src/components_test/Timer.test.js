
import React, { Component } from 'react';
import Timer from "../components/client_components/Timer"
import { shallow } from 'enzyme'


describe('UI Test for timer', () => {

    it('Timer to be defined', () => {
        expect(Timer).toBeDefined();
    });

    it('Timer should render correctly', () => {
        const snapshot = shallow(
            <Timer />
        );
        expect(snapshot).toMatchSnapshot();
    });

    it('Timer should stop after specified time', () => {
        var specifiedTime = 10;
        const wrapper = shallow(
            <Timer seconds={specifiedTime} />
        );
        setTimeout(function () {
            expect(wrapper.state('timer')).toEqual(0);
        }.bind(this), specifiedTime * 100);
    });
    it('Timer text should be equal to seconds left', () => {
        var specifiedTime = 10;
        const wrapper = shallow(
            <Timer seconds={specifiedTime} />
        );
        setTimeout(function () {
            expect(wrapper.find('a').text()).toEqual("00 : 05");
        }.bind(this), (specifiedTime/2) * 100);
    });
    it('Timer text should show 00 : 00 after timer is out', () => {
        var specifiedTime = 10;
        const wrapper = shallow(
            <Timer seconds={specifiedTime} />
        );
        setTimeout(function () {
            expect(wrapper.find('a').text()).toEqual("00 : 00");
        }.bind(this), specifiedTime * 100);
    });
    it('Timer text after timeout', () => {
        var specifiedTime = 10;
        const wrapper = shallow(
            <Timer seconds={specifiedTime} />
        );
        setTimeout(function () {
            expect(wrapper.find('a').text()).toEqual("00 : 00");
        }.bind(this), (specifiedTime+3) * 100);
    });
    it('Timer text should be equal to specified time at the start', () => {
        var specifiedTime = 10;
        const wrapper = shallow(
            <Timer seconds={specifiedTime} />
        );
        expect(wrapper.find('a').text()).toEqual(`00 : ${specifiedTime}`);
    });
})