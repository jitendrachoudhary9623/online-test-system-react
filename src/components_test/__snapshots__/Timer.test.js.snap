exports[`UI Test for timer Timer should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Timer
    seconds={60} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="navbar-fixed waves-effect waves-light btn grey darken-5">
        01
         : 
        00
</a>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "01",
          " : ",
          "00",
        ],
        "className": "navbar-fixed waves-effect waves-light btn grey darken-5",
      },
      "ref": null,
      "rendered": Array [
        "01",
        " : ",
        "00",
      ],
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="navbar-fixed waves-effect waves-light btn grey darken-5">
          01
           : 
          00
</a>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "01",
            " : ",
            "00",
          ],
          "className": "navbar-fixed waves-effect waves-light btn grey darken-5",
        },
        "ref": null,
        "rendered": Array [
          "01",
          " : ",
          "00",
        ],
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
